fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/PartitionColumnMinMaxRewriteRule.java

```diff
@@ -0,0 +1,362 @@
+import java.util.stream.Collectors;
+
+/**
+ * Optimization rule for MIN/MAX PARTITION_COLUMN:
+ * 1. Partition Prune: only choose the largest partition for MAX(col)
+ * 2. Partition Values: evaluate the MAX(col) in optimizer for LIST-PARTITION
+ * 3. TopN: transform the MAX(col) into TopN query
+ */
+public class PartitionColumnMinMaxRewriteRule extends TransformationRule {
+
+    public PartitionColumnMinMaxRewriteRule() {
+        super(RuleType.TF_PARTITION_COLUMN_MINMAX, Pattern.create(OperatorType.LOGICAL_AGGR)
+                .addChildren(Pattern.create(OperatorType.LOGICAL_PROJECT, OperatorType.LOGICAL_OLAP_SCAN)));
+    }
+
+    @Override
+    public boolean check(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggregationOperator = input.getOp().cast();
+        LogicalOlapScanOperator scanOperator = input.getInputs().get(0).getInputs().get(0).getOp().cast();
+        Table table = scanOperator.getTable();
+        return context.getSessionVariable().isEnableRewritePartitionColumnMinMax()
+                && table.isNativeTableOrMaterializedView()
+                && ((OlapTable) table).isPartitionedTable()
+                && checkMinMaxAggregation(aggregationOperator, scanOperator, context);
+    }
+
+    /**
+     * Check simple aggregation functions like MIN/MAX
+     * 1. No GROUP-BY
+     * 2. No HAVING
+     * 3. Only MAX or MAX
+     * 4. MIN/MAX only refers partition-column
+     */
+    private boolean checkMinMaxAggregation(LogicalAggregationOperator aggregationOperator,
+                                           LogicalScanOperator scanOperator,
+                                           OptimizerContext context) {
+        boolean simpleAgg = aggregationOperator.getAggregations().entrySet().stream().allMatch(
+                entry -> {
+                    CallOperator aggregator = entry.getValue();
+                    AggregateFunction aggregateFunction = (AggregateFunction) aggregator.getFunction();
+                    String functionName = aggregateFunction.functionName();
+                    ColumnRefSet usedColumns = aggregator.getUsedColumns();
+                    if (functionName.equals(FunctionSet.MAX) || functionName.equals(FunctionSet.MIN)) {
+                        if (usedColumns.size() != 1) {
+                            return false;
+                        }
+                        ColumnRefOperator usedColumn =
+                                context.getColumnRefFactory().getColumnRef(usedColumns.getFirstId());
+                        Column column = scanOperator.getColRefToColumnMetaMap().get(usedColumn);
+                        if (column == null) {
+                            // not a colum on table
+                            return false;
+                        }
+                        OlapTable olapTable = (OlapTable) scanOperator.getTable();
+                        if (!olapTable.getPartitionColumns().contains(column)) {
```

Comment:
What about table with multiple partition columns?

Reply:
not supported